checkout/views.py

```diff
@@ -0,0 +1,148 @@
+                    except stripe.error.CardError:
+                        messages.error(request, "Your card has been declined.")
+
+                    if customer.paid:
+                        for item in orders:
+                            item.is_paid = True
+                            item.order_date = date.today()
+                            if user_answer == comp.correct_answer:
+                                item.user_answer_correct = True
+                            item.save()
+
+                        user_correct = False
+                        if user_answer == comp.correct_answer:
+                            user_correct = True
+                            for item in orders:
+                                tickets_per_order = item.quantity
+                                while tickets_per_order > 0:
+                                    create = True
+                                    while create:
+                                        ticket_number = randint(1, comp.tickets)
+                                        entry, created = Entries.objects.get_or_create(
+                                            defaults={
+                                                'user': user,
+                                                'order': item
+                                            },
+                                            competition_entry=comp,
+                                            ticket_number=ticket_number
+                                        )
+                                        if created:
+                                            tickets_per_order -= 1
+                                            create = False
+
+                            tickets_left = comp.tickets_left
+                            comp.tickets_left = tickets_left - tickets
+                        comp.save()
+                        email_order(request, orders, total, user_correct)
+                        request.session['user_correct'] = user_correct
+                        if comp.tickets_left < 500:
+                            try:
+                                Competition.objects.get(
+                                    next_competition=True
+                                )
+                            except Competition.DoesNotExist:
+                                new_competition()
+
+                        if comp.tickets_left == 0:
+                            pick_competition_winner()
+                        return redirect('checkout_complete')
+                else:
+                    messages.error(
+                        request,
+                        "We are unable to take payment from that card."
+                    )
+    else:
+        payment_form = PaymentForm()
+
+    content = {
+        'user': request.user,
+        'orders': orders,
+        'comp': comp,
+        'payment_form': payment_form
+    }
+    return render(request, 'checkout.html', content)
```

Comment:
This function is too long, ideally functions should not be more than 50 lines long. Abstract some functionality into other utility functions